clients/da-vinci-client/src/main/java/com/linkedin/davinci/blobtransfer/client/P2PFileTransferClientHandler.java

```diff
@@ -129,13 +170,45 @@ protected void channelRead0(ChannelHandlerContext ctx, HttpObject msg) throws Ex
               "File size mismatch for " + fileName + ". Expected: " + fileContentLength + ", Actual: "
                   + outputFileChannel.size());
         }
-        resetState();
       }
     } else {
       throw new VeniceException("Unexpected message received: " + msg.getClass().getName());
     }
   }
 
+  /**
+   * Handles the metadata transfer response.
+   * @param ctx the channel context
+   * @param msg the http object
+   * @throws JsonProcessingException
+   */
+  private BlobTransferPartitionMetadata handlerMetadataTransferResponse(ChannelHandlerContext ctx, HttpObject msg)
+      throws JsonProcessingException {
+    if (msg instanceof HttpContent) {
```

Comment:
I think we are pretty sure the metadata has to be `HttpContent` in order to process it so instead of doing the check here, I'd suggest we update the method signature to be ` handlerMetadataTransferResponse(ChannelHandlerContext ctx, HttpContent msg)`